File: 2378/2378.c
// bee 2378 - Elevador
// Alonso Martins
// 19/05/2024
#include <stdio.h>

int main()
{
    int N, M, S, E, p = 0; scanf("%d %d", &N, &M);

    while (N -- && scanf("%d %d", &S, &E))
    {
        p -= S, p += E;
        if (p > M)
        {
            printf("S\n");
            return 0;
        }
    }

    printf("N\n");
    return 0;
}

File: 2378/2378.py
# bee 2378 - Elevador
# Alonso Martins
# 19/08/2024
N, capacidadeMaxima = map(int, input().split())
noElevador = 0
ultrapassou = False

while N:
    o, i = map(int, input().split())
    noElevador += i - o

    if noElevador > capacidadeMaxima:
        ultrapassou = True
        break

    N -= 1

print("S" if ultrapassou else "N")
